README.md

```diff
@@ -122,11 +122,39 @@ LOG_LEVEL=trace npm run test
 
 Alternatively, one may set `LOG_LEVEL` in the `.env` file.
 
-### Authentication
-
-The service validates a Bearer JSON Web Token (JWT) found in the `Authorization` HTTP header.
-
-See the `.env.example` for relevant variables.
+### Authentication and Authorization
+
+A valid Bearer JSON Web Token (Bearer JWT) is required in requests to the PDC service. The PDC officially reccomends using [KeyCloak](https://www.keycloak.org/) as the authentication provider. Please refer to [the relevant documentation for setup and configuration](https://www.keycloak.org/guides#getting-started)
+
+The relevant authorization enviroment variables as they appear in the `.env.example` file are:
+
+- `AUTH_SERVER_ISSUER`: the URL for the issuer (this is the "realm" URL in Keycloak terminology).
+- `OPENAPI_DOCS_AUTH_CLIENT_ID`: the public identifier for the app ("Client ID" field in Keycloack).
+
+#### An example with a Keycloak authentication and authorization service
+
+From the Keycloak admin interface, e.g. https://my-host/admin:
+
+1. Add a realm e.g. "pdc" (avoid spaces in the name).
+2. Within the realm just added in step (1) add a client e.g. "pdc-service", with Client authentication off and Authorization off:
+   - Standard flow unchecked,
+   - Direct access grants unchecked,
+   - Implicit flow unchecked,
+   - Service accounts roles unchecked,
+   - OAuth 2.0 Device Authorization Grant (optional), and
+   - OIDC CIBA Grant unchecked.
+3. In the settings for the client added in step (2), set the root URL and Home URL to the URL of the PDC service (not the auth service). Add redirect URIs:
+   - `/applicants/*`
+   - `/applications/*`
+   - `/applicationForms/*`
+   - `/canonicalFields/*`
+   - `/proposals/*`
+   - `/proposalVersions/*`
+4. In the settings for the client added in step (2), add a Web origins of `+`.
+5. With in the realm, add a user, e.g. `test-user`.
+6. Log in as the test user.
```

Comment:
What is the use case in mind for steps 6, 7, and 8?

If the use case is a developer making API calls, they could use the swagger-generated page (e.g. [https://api.philanthropydatacommons.org](https://api.philanthropydatacommons.org)) which handles auth for them.  If the use case is a script or application making API calls then we don't want them to have to manually log in as a human in order to interact with the API, and instead we should provide instructions on creating a new client for their application which gives them a token / secret pair that they can use to avoid an auth flow that requires manual log in.